keps/349-avoid-coscheduling/README.md

```diff
@@ -0,0 +1,445 @@
+
+If the workload fails to schedule its pods it could block admission of other
+workloads indefinitely.
+
+To mitigate this issue we introduce introduce a timeout on reaching the `PodsReady`
+condition by a workload since its job start (see:
+[Timeout on reaching the PodsReady condition](#timeout-on-reaching-the-podsready-condition)).
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+
+### PodsReady workload condition
+
+First, we introduce a new workload condition, called `PodsReady`, to indicate
+that the first batch of a workload's pods is scheduled, more precisely:
+`job.status.ready + job.status.succeeded >= job.spec.parallelism`.
+
+The `PodsReady` condition is addded to the workload by the Kueue's Job
+reconciler in reaction to a status update of the corresponding Job. Note that,
+verifying if the condition should be added does not require an extra API call as
+the Kueue's Job reconciller already fetches the latest Job object at the
+beginning of the `Reconcile` function. Also note, there is no new condition
+introduced to the Job API.
+
+### Workload API
+
+We extend the Workload API with the new `startupRequirements` field to
+enable the new behavior.
+
+```golang
+// StartupRequirements describes workload startup requirements
+type StartupRequirements string
+
+const (
+  // QuotaAndProdsReady means that workload requires all pods to be running at
+  // the same time
+  StartupRequirementsQuotaAndProdsReady StartupRequirements = "QuotaAndProdsReady"
+  // QuotaOnly means that the only requirement for a workload to start is to
+  // get resources from the cluster quota
+  StartupRequirementsQuotaOnly StartupRequirements = "QuotaOnly"
+)
+
+// WorkloadSpec defines the desired state of Workload
+type WorkloadSpec struct {
+  ...
+  // StartupRequirements describes the workload requirement on the resource allocation.
+  // Possible values are:
+  // - QuotaAndProdsReady: means that workload requires allocation of resources
+  //   to support running of all pods at the same time,
+  // - QuotaOnly: means that the workload does not require all pods to be running
+  //   at the same time.
+  // Defaults to QuotaOnly.
+  StartupRequirements *string `json:"startupRequirements,omitempty"`
+}
+```
+
+### Job annotation
+
+We introduce a new Job annotation, called `kueue.x-k8s.io/workload-startup-requirements`,
```

Comment:
```suggestion
We introduce a new Job annotation, called `kueue.x-k8s.io/startup-requirements`,
```

Workload is mostly an internal concept that people that right Jobs don't need to know about.

Reply:
Adjusted